src/Raven.Server/Documents/CountersStorage.cs

```diff
@@ -518,6 +520,23 @@ public string PutOrIncrementCounter(DocumentsOperationContext context, string do
             }
         }
 
+        private static void CheckForCorruptedCountersData(string documentId, BlittableJsonReaderObject counterValues, BlittableJsonReaderObject counterNames)
+        {
+            // RavenDB-22835
+            // counters data might be corrupted and needs to be fixed before adding/incrementing new counters
+
+            var msg = $"Counters data of document '{documentId}' is corrupted. " +
```

Comment:
you are creating that each time, even when no exception is thrown. I would suggest to create a local method that throws with this message, this will avoid the issue